Document grouped results layout and CATEGORIZE patterns in Discover ES|QL (#6190)
elastic/docs-content: explore-analyze/discover/try-esql.md

%
% When the query also computes a [`SPARKLINE`](elasticsearch://reference/query-languages/esql/functions-operators/aggregation-functions/sparkline.md) over time, the resulting array is rendered as an inline sparkline next to the row aggregates. For example, the following query categorizes log messages and renders a per-pattern sparkline:
%
% ```esql
% FROM kibana_sample_data_logs
% | WHERE @timestamp <= ?_tend AND @timestamp > ?_tstart
% | STATS Count = COUNT(*),
%         Sparkline = SPARKLINE(COUNT(*), @timestamp, 40, ?_tstart, ?_tend)
%     BY Pattern = CATEGORIZE(message)
% | SORT Count DESC
% ```
%
% On larger data sets, add a [`SAMPLE`](elasticsearch://reference/query-languages/esql/commands/sample.md) command before `STATS` to keep the categorization fast, and divide `COUNT(*)` by the same sample fraction to keep the counts representative. For example, `SAMPLE 0.001` followed by `Count = COUNT(*) / 0.001`.
%
% :::{image} /explore-analyze/images/discover-esql-cascade-pattern-sparkline.png
% :alt: A grouped row showing a CATEGORIZE pattern with token highlighting and an inline sparkline
% :screenshot:
% :::
% RESTORE FOR 9.5 - end

::::{tip}

> Pattern detection on text fields is also available outside {{esql}} from the **Patterns** tab in Discover's classic mode. See [](/explore-analyze/discover/run-pattern-analysis-discover.md).

Reviewer: ```suggestion
Pattern detection on text fields is also available outside {{esql}} from the **Patterns** tab in Discover's classic mode. Refer to [](/explore-analyze/discover/run-pattern-analysis-discover.md).
```

::::

### Grouped row actions

Select the {icon}`boxes_vertical` actions button on any group row to:

- **Copy to clipboard**: copy the group's value.
- **Filter in**: append a `WHERE` clause to your query that keeps only documents matching this group.
- **Filter out**: append a `WHERE` clause that excludes documents matching this group.
- **Open in new tab**: open the documents in this group in a new Discover tab, with a query scoped to that group.

**Filter in** and **Filter out** are disabled when the grouping field is not filterable.

### Opt out of the grouped layout

> When the grouped layout activates, the regular results table toolbar is replaced with a {icon}`flask` **Group by** button. The button shows the number of active groupings as a badge, and hovering it confirms that grouped results are a technical preview.

Reviewer: ```suggestion
When the grouped layout activates, the regular results table toolbar is replaced with a {icon}`flask` **Group by** button. The button shows the number of active groupings as a badge, and hovering over it confirms that grouped results are in technical preview.
```

Author: ```suggestion
When the grouped layout activates, the regular results table toolbar is replaced with a {icon}`flask` **Group by** button. The button shows the number of active groupings as a badge.
```
